File: jsExercises/lasagna.js
export const MY_VALUE = 20;
export const EXPECTED_MINUTES_IN_OVEN = 40;
const PREPARATION_MINUTES_PER_LAYER = 2;

export function add(a, b) {
	return a + b;
}

export function remainingMinutesInOven(actualMinutesInOven) {
	return EXPECTED_MINUTES_IN_OVEN - actualMinutesInOven;
}

// Each Layer Takes 2 Minutes to Prepare
export function preparationTimeInMinutes(numberOfLayers) {
	return numberOfLayers * PREPARATION_MINUTES_PER_LAYER;
}

export function totalTimeInMinutes(numberOfLayers, actualMinutesInOven) {
	return add(preparationTimeInMinutes(numberOfLayers), actualMinutesInOven);
}

console.log(totalTimeInMinutes(3, 20));

File: javaExercises/Lasagna.java
public class Lasagna {


    public int expectedMinutesInOven() {
        return 40;
    }

    public int remainingMinutesInOven(int actualMinutesInOven) {
        return expectedMinutesInOven() - actualMinutesInOven;
    }

    public int preparationTimeInMinutes(int layers) {
        // Each layer takes 2 minutes to cook
        return 2 * layers;
    }

    public int totalTimeInMinutes(int layers, int numMinutesInOven) {
        return preparationTimeInMinutes(layers) + numMinutesInOven;
    }
    
    public static void main(String[] args) {
        Lasagna lasagna = new Lasagna();
        int output = lasagna.preparationTimeInMinutes(30);
        System.out.println(output);
    }
}